src/Bathymetry.jl

```diff
@@ -115,12 +90,9 @@ Keyword Arguments
                   the smallest basins are removed first. `major_basins = 1` retains only the largest basin.
                   If `Inf` then no basins are removed. Default: 1.
 """
-function regrid_bathymetry(target_grid;
+function regrid_bathymetry(target_grid, m::Metadatum;
```

Comment:
It is quite inconvenient to have to build a `Metadatum` object to download the bathymetry when (generally speaking) the only information you need is the dataset. Let's add a method like:
```julia
function regrid_bathymetry(target_grid; dataset = ETOPOBathymetry(), kw...)
    metadatum = # Build metadata object...
    return regrid_bathymetry(target_grid, metadatum; kw...)
end
```

Reply:
Actually, this is good point. Also, if we add a method like this then this PR won't be introducing breaking API changes since all examples would continue to work without changes. Then we can change the bump minor release to just bump a patch release.

Reply:
Also, btw, we need to update the docstring of this method.

Reply:
I don't understand:

1) `regrid_bathymetry` is no longer used to "download the bathymetry" - that is now in `download_dataset`. 
2) We will still need to redefine `regrid_bathymetry`, just with `dataset` as an added kwarg instead of `metadatum` - so we will still introduce breaking changes for all examples that load a bathymetry. 
3) I am happy to switch `metadata` with `dataset` in the `regrid_bathymetry`, though I am unsure of what the added benefit of this would be - simply that we are ensuring the user doesn't have to predefined their metadata? 
4) In my opinion, letting the user predefined metadatum first is helpful, because then they can at least inspect the metadatum, like the directory, etc.

Reply:
@simone-silvestri was suggesting the above method as an _additional_ convenience method. He's not suggesting we drop the method with the metadadum.

I implemented this in 

https://github.com/CliMA/ClimaOcean.jl/blob/6bb3984272140e907b217acdd89c9adc27d16f47/src/Bathymetry.jl#L143-L147

Reply:
> regrid_bathymetry is no longer used to "download the bathymetry" - that is now in download_dataset.

hm... ok, I might have missed that. I think similarly to other datasets, download should be also included in methods that use the data (eg in `set!`). This way eg regrid_bathymetry will download the data if it's not there. But regrid_bathymetry _won't be the only way_ to download the data.